assets/js/googlesitekit/datastore/user/nonces.js

```diff
@@ -0,0 +1,175 @@
+
+const fetchGetNoncesStore = createFetchStore( {
+	baseName: 'getNonces',
+	controlCallback: () => {
+		return API.get( 'core', 'user', 'nonces', undefined, {
+			useCache: false,
+		} );
+	},
+	reducerCallback: ( state, nonces ) => ( {
+		...state,
+		nonces,
+	} ),
+} );
+
+const baseInitialState = {
+	nonces: null,
+};
+
+const baseActions = {
+	/**
+	 * Sets nonces.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} nonces User capabilities.
+	 * @return {Object} Redux-style action.
+	 */
+	receiveNonces( nonces ) {
+		return {
+			type: RECEIVE_NONCES,
+			payload: { nonces },
+		};
+	},
+};
+
+const baseControls = {};
+
+const baseReducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case RECEIVE_NONCES: {
+			const { nonces } = payload;
+
+			return {
+				...state,
+				nonces,
+			};
+		}
+
+		default: {
+			return state;
+		}
+	}
+};
+
+const baseResolvers = {
+	*getNonces() {
+		const registry = yield Data.commonActions.getRegistry();
+
+		const existingNonces = yield registry.select( CORE_USER ).getNonces();
+		// if existingNonces is not empty array
+		if ( existingNonces ) {
+			// console.log( 'has nonces', existingNonces );
+			return existingNonces;
+		}
+
+		// Temporary hack to preserve previous behavior when resolved from global.
+		// This is necessary for now to avoid a delay if the preloaded data
+		// has already expired.
+		const preloadedNonces =
+			global._googlesitekitAPIFetchData?.preloadedData?.[
+				'/google-site-kit/v1/core/user/data/nonces'
+			]?.body;
+
+		if ( preloadedNonces ) {
+			yield fetchGetNoncesStore.actions.receiveGetNonces( {
+				...preloadedNonces, // dereference.
+			} );
+		}
+
+		if ( ! preloadedNonces ) {
+			// console.log( 'no nonces', existingNonces );
+			const { response } = fetchGetNoncesStore.actions.fetchGetNonces();
+
+			if ( response ) {
+				yield fetchGetNoncesStore.actions.receiveGetNonces( {
+					...response, // dereference.
+				} );
+			} else {
+				return null;
+			}
+		}
+	},
+};
+
+const baseSelectors = {
+	/**
+	 * Gets capabilities of the current user.
+	 *
+	 * @since 1.13.0
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(Object|undefined)} Capabilities object. Returns undefined if it is not loaded yet.
+	 */
+	getNonces( state ) {
+		const { nonces } = state;
+		return nonces;
+	},
+
+	/**
+	 * Checks if the current user has the specified capability or not.
+	 *
+	 * @since 1.13.0
+	 *
+	 * @param {Object} state Data store's state.
+	 * @param {string} nonce Nonce action to check.
+	 * @param {Array}  args  List of rest of the arguments. Specifically one or many module slugs.
+	 * @return {(boolean|undefined)} TRUE if the current user has this capability, otherwise FALSE. If capabilities ain't loaded yet, returns undefined.
+	 */
+	getNonce: createRegistrySelector( ( select ) => ( state, nonce ) => {
+		const nonces = select( CORE_USER ).getNonces();
+
+		if ( nonces ) {
+			return nonces[ nonce ];
+		}
+
+		return undefined;
```

Comment:
A minor point, but we often try to "return early" rather than have the fall-through be `undefined`.

```suggestion
		if ( nonces === undefined ) {
			return undefined
		}

		return nonces[ nonce ];
```

That just makes it clear what the "ideal" return value will be and why it would return something _other_ than that. You could do:

```jxs
		return nonces?.[ nonce ];
```

but I actually think the explicit `return undefined` is clearly in its intention 🙂